ee/clickhouse/client.py

```diff
@@ -130,40 +131,58 @@ def cache_sync_execute(query, args=None, redis_client=None, ttl=CACHE_TTL, setti
             redis_client.set(key, _serialize(result), ex=ttl)
             return result
 
+    InsertParams = Union[list, tuple, types.GeneratorType]
+    NonInsertParams = Union[Dict[str, Any]]
+    QueryArgs = Optional[Union[InsertParams, NonInsertParams]]
+
+    def prepare_query(client: SyncClient, query: str, args: QueryArgs):
```

Comment:
Nit: Let's order methods in order of importance here. `sync_execute` is the only function to be used outside of this file. Everything else should be _after_ that is possible, it's worth prepending a `_` to signal this is not "official" functionality either.